packages/api-gateway/src/services/catalogService.ts

```diff
@@ -259,5 +259,6 @@ export async function enhanceEservice(
     isSignalHubEnabled: eservice.isSignalHubEnabled,
     isConsumerDelegable: eservice.isConsumerDelegable,
     isClientAccessDelegable: eservice.isClientAccessDelegable,
+    templateRef: eservice.templateRef,
```

Comment:
The output of this `enhance` method is a mix of info from the e-service and it's latest version (latest descriptor). For this reason, I would also include the template version from the descriptor ref. Something like:

```suggestion
    templateRef: {
       templateId: eservice.templateRef.id,
       instanceId: eservice.templateRef.instanceId,
       templateVersionId: latestValidDescriptor.templateVersionRef.id
     }
```
Adapt API spec as needed.

Reply:
@ecamellini 
To be able to add **templateVersionId**, I enabled the possibility of having additional fields inside **templateRef** in the API spec.  

I'm not sure if this is the usual approach, or if it would be better to explicitly define upfront that **templateRef** can (optionally) include this specific field.